### PR TITLE
VIM-4858-Removing-Extra-Percent-Encoding

### PR DESCRIPTION
**Pull request by supperspidey**
#### Ticket
[VIM-4858](https://vimean.atlassian.net/browse/VIM-4858)

#### Ticket Summary
`NSURLComponents` encoded percent signs for a query that was already percent-encoded. For example, if my query string is "My%20name%20is%20Van" &mdash; which is percent-encoded &mdash; then the class will turn the string into "My%2520name%2520is%2520Van" because the `%` sign is encoded as `%25`.

#### Implementation Summary
Since the query string passed into `NSURLComponents`, when we get the string out of the class, we need to undo the extra percent encoding. We can achieve that by sending a `stringByRemovingPercentEncoding` message to the `components.string` property.

#### How to Test
1. Tap on the Search icon.
2. Input a query string with a space character in it.

### CODE REVIEW

File: VimeoNetworking/Sources/Request.swift [hunk 1]
```diff
@@ -181,7 +181,14 @@ public struct Request<ModelType: MappableResponse>
...
         
-        return components!.string!
+        if let theURI = components?.string?.stringByRemovingPercentEncoding
+        {
+            return theURI
+        }
+        else
+        {
+            return ""
+        }
```
Comment (jenniferlim) [inline VimeoNetworking/Sources/Request.swift]: Nit - How about `return components.string?.stringByRemovingPercentEncoding ?? ""`

File: VimeoNetworking/Sources/Request.swift [hunk 2]
```diff
@@ -181,7 +181,14 @@ public struct Request<ModelType: MappableResponse>
             }
         }
         
-        return components!.string!
+        if let theURI = components?.string?.stringByRemovingPercentEncoding
```
Comment (jenniferlim) [inline VimeoNetworking/Sources/Request.swift]: Nit - Should we `guard` at the top of the functions? So this `?` on `components` would disappear 💭

**Comment by jenniferlim** [comment]
@supperspidey - I just have 2 minor comments. Otherwise looks 🌟.

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/4b9b46cf0899537379d4e38f240478df0fa474ea/VimeoNetworking/Sources/VimeoClient.swift#L322">VimeoNetworking/Sources/VimeoClient.swift#L322</a></strong>: cast from ‘Response&lt;VIMNullResponse&gt;’ to unrelated type ‘Response&lt;ModelType&gt;’ always fails  <br /><code>
                let response = Response(model: nullResponseObject, json: [:]) as! Response&lt;ModelType&gt;
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 22 tests, with 0 failures (0 unexpected) in 0.264 (0.274) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/113)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by jenniferlim** [comment]
@supperspidey - 👍 👍 👍

**Pull request merged**